### DISCUSSION TITLE
error: expected one of `->`, `where`, or `{`, found `<eof>`

### DISCUSSION THREAD

**Discussion by Caligino** [bevyengine/bevy#6405] [Q&A]
I have a derive statement at the end of my file that appears to be correct and was working prior to a revision to my program, but afterwards the compiler is throwing an error. Note that my program references an external rust file as a drop in library.

```#[derive(Clone, Eq, PartialEq, Debug, Hash)]
enum MyStates {
    AssetLoading,
    Next,
}
```

The error is on the closing brace, it is the only error the compiler has detected.  

``` 
^ expected one of `->`, `where`, or `{`
```
It wants a return value for something apparently, but i'm not where in the code it's required.

Note that this library is the orbitcamera library so it's possible that there's an unsatisfied trait somewhere that need to be fulfilled, but the compiler isn't picking up on it.

**Comment by MrGVSV** [comment]
Could you share a bit more code? Are you deriving a custom trait or one of Rust's built-ins?
